code/datums/traits/_quirk.dm

```diff
@@ -4,120 +4,135 @@
 	var/name = "Test Quirk"
 	var/desc = "This is a test quirk."
 	var/value = 0
-	var/human_only = TRUE
+	var/mob/living/restricted_type = /mob/living/carbon/human //specifies which mob subtype can use this trait
```

Comment:
I think this should be list type, not a single variable. If I say an example, there can be a quirk like "Overoxygionated" that can increase the sprite size of moth/apid/and flyperson characters. (as insects with a lot of oxygen can increase their size in real), but it won't be nothing to non-insects.

There is an example of such case as `restricted_species = list(SPECIES_LIZARD)` in `uplink_items.dm`